interfaces/system_key.go

```diff
@@ -33,34 +38,68 @@ import (
 type systemKey struct {
 	BuildID          string   `yaml:"build-id"`
 	AppArmorFeatures []string `yaml:"apparmor-features"`
+	NFSHome          bool     `yaml:"nfs-home"`
+	Core             string   `yaml:"core,omitempty"`
 }
 
+var mockedSystemKey *systemKey
+
 func generateSystemKey() *systemKey {
-	var mySystemKey systemKey
+	// for testing only
+	if mockedSystemKey != nil {
+		return mockedSystemKey
+	}
 
-	// add build-id
+	var sk systemKey
 	buildID, err := osutil.MyBuildID()
 	if err != nil {
-		buildID = "unknown"
+		buildID = ""
 	}
-	mySystemKey.BuildID = buildID
+	sk.BuildID = buildID
+
+	// Add apparmor-feature (which is already sorted)
+	sk.AppArmorFeatures = release.AppArmorFeatures()
 
-	// Add apparmor-feature, note that ioutil.ReadDir() is already sorted.
+	// Add if home is using NFS, if so we need to have a different
+	// security profile and if this changes we need to change our
+	// profile.
+	sk.NFSHome, err = osutil.IsHomeUsingNFS()
+	if err != nil {
+		logger.Noticef("cannot determine nfs usage in generateSystemKey: %v", err)
+	}
+	// Add the current Core path, we need this because we call helpers
+	// like snap-confine from core that will need an updated profile
+	// if it changes
 	//
-	// We prefix the dirs.GlobalRootDir (which is usually "/") to make
-	// this testable.
-	mySystemKey.AppArmorFeatures = release.AppArmorFeatures()
+	// FIXME: what about core18? the snapd snap?
```

Comment:
Can you expand this comment? What do you mean, what's the problem?

Reply:
I think it's that the same reasoning (we call helpers from core) might apply to the snapd snap, or to core18.